GetStream/stream-chat-swift
# [CIS-1210, CIS-1224] Fix message list jumps

### 🔗 Issue Link
CIS-1210, CIS-1224

### 🎯 Goal
Fix the message list jump UI Glitch when a new reaction is added and the message is partially hidden by the composer.

When a message is partially hidden by the composer, and the message cell is reloaded, the message list does a weird jump. Possibly this happens because the cell is out of scope of the message list, and we can see the message cell calculating the height of the cell, and initially, it renders it as zero.

**BONUS:** This fixes also messages list jumping when a new message is sent and as a user you are scrolled up (Regression).

### 🛠 Implementation
~In order to fix this, we need to provide a cell height estimation on the table view and cache the message cell's height before the message cell is reloaded.~

Update: Caching the message cell's height caused more serious problems, especially when sending messages where the previous messages were too big. Plus, the issue was not completely gone, there was still a noticeable jump on the message list. For this reason, right now we disable the animations when reloading a cell to avoid this. Until we move to a collection view, there isn't much we can do to improve this.

### 🧪 Testing
- Cover half a message below the composer
- Make the message reload (Add a reaction to a message without reactions, so the cell type changes)
- Observe that the table view doesn't jump.

### 🎨 Changes
| Before | After |
| --- | --- |
| <video src="https://user-images.githubusercontent.com/12814114/137034678-a21a061f-a192-496b-92af-2ed3e29cca80.mp4"> | <video src="https://user-images.githubusercontent.com/12814114/137166078-c40edd4b-252d-4943-a2cf-eee225c56b27.mp4"> |


### ☑️ Checklist
